instrumentation/async-http-client-1.9/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/ResponseAdvice.java

```diff
@@ -26,14 +25,14 @@ public static Scope onEnter(
     // After response was handled by user provided handler.
     ContextStore<AsyncHandler, Pair> contextStore =
         InstrumentationContext.get(AsyncHandler.class, Pair.class);
-    Pair<Context, Span> spanWithParent = contextStore.get(handler);
-    if (null != spanWithParent) {
+    Pair<Context, Context> contextWithParent = contextStore.get(handler);
+    if (contextWithParent != null) {
       contextStore.put(handler, null);
     }
-    if (spanWithParent.hasRight()) {
-      AsyncHttpClientTracer.tracer().end(spanWithParent.getRight(), response);
+    if (contextWithParent.hasRight()) {
+      AsyncHttpClientTracer.tracer().end(contextWithParent.getRight(), response);
     }
-    return spanWithParent.hasLeft() ? spanWithParent.getLeft().makeCurrent() : null;
+    return contextWithParent.hasLeft() ? contextWithParent.getLeft().makeCurrent() : null;
```

Comment:
I think this is very suspicious. Why we make the context, which were active before http call on some other thread, active on response thread? Probably a separate issue from this PR but still

Reply:
this existing behavior seems good to me, so that any callbacks will get associated to the request instead of lost in the void